kratos/includes/key_hash.h

```diff
@@ -90,6 +90,29 @@ namespace Kratos
 ///@{
 
     class VariableData; // forward declaration
+    
+    /**
+     * @brief A custom hash function for std::vector<TIntType>.
+     * @details This struct defines a custom hash function for std::vector<TIntType>. It computes the hash value of a vector by iterating through its elements and combining their hash values.
+     * @tparam TIntType The integer type
+     */
+    template<class TIntegerType = int>
+    struct VectorIntegerHash 
```

Comment:
Jus curious, why call this `VectorIntegerHash` when it can be done with any `std::vector<Type>`?

Reply:
Nope, because the only type which is a hash itself is an integer, other type should be cast to integer and that may not always work